### ISSUE TITLE
Link Exception in 4.0.0-alpha.10

### ISSUE THREAD

**Issue by aseef17** [zino-app/graphql-flutter#734]
**Describe the issue**
When trying to pass custom objects to variables, we get the following error

LinkException: LinkException('package:gql_http_link/src/_utils.dart': Failed assertion: line 75 pos 5: 'body is String || body is num || body == null'

**Expected behavior**
Custom objects should be allowed in variables. This worked fine in V 3.1.0

**device / execution context**
Are you on iOS, android, web, in a simulator, running from the terminal etc? This is especially important for `localhost` connection issues.
No.

<details>
  <summary>Stacktrace:</summary>

```dart
LinkException: LinkException('package:gql_http_link/src/_utils.dart': Failed assertion: line 75 pos 5: 'body is bool || body is String || body is num || body == null': Instance of 'OrderProduct' of type OrderProduct was found in in the request at path variables.products.0, but the only the types { Map, List, MultipartFile, bool, String, num, null } are allowed)
```

**Comment by micimize** [comment]
Right you are! Got rid of that assertion and should be solved in `4.0.0-alpha.11`

**Comment by aseef17** [comment]
> Right you are! Got rid of that assertion and should be solved in `4.0.0-alpha.11`

Thanks for the quick response. I'll test it out and close the issue if it's fixed
